### PR TITLE
Add multiplier option to prometheus helper

### PR DESCRIPTION
**Pull request by jsoriano**
Prometheus histograms can contain buckets with keys in decimal
representation. They create events with dots in the field names, what
can be problematic when stored in Elasticsearch as they will create
sub-objects.

Add a new option to allow to store these keys in different units, so
precission is kept but in a different order of magnitude.

### CODE REVIEW

**Comment by jsoriano** [comment]
jenkins, test this again please

**Comment by ChrsMark** [comment]
Hello, any update on this?

File: metricbeat/helper/prometheus/metric.go [hunk 1]
```diff
@@ -57,6 +57,13 @@ func OpLowercaseValue() MetricOption {
 	return opLowercaseValue{}
 }
 
+// OpMultiplyBuckets multiplies bucket labels in histograms, useful to change units
+func OpMultiplyBuckets(multiplier float64) MetricOption {
```
Comment (sayden) [inline metricbeat/helper/prometheus/metric.go]: I'd call this `NewOpMultiplyBucketsOption` unless we have some internal convention for this

Comment (jsoriano) [reply metricbeat/helper/prometheus/metric.go]: I don't have an strong opinion on this but yes, I am following the name schemas used here.

Take into account that they are not so bad as they are used as options in prometheus metrics:
```
Metrics: map[string]MetricMap{
    "histogram_decimal_metric": Metric("histogram.metric", OpMultiplyBuckets(1000))
}
```

File: metricbeat/helper/prometheus/metric.go [hunk 2]
```diff
@@ -267,3 +274,30 @@ func (o opLowercaseValue) Process(field string, value interface{}, labels common
 	}
 	return field, value, labels
 }
+
+type opMultiplyBuckets struct {
+	multiplier float64
+}
+
+// Process will multiply the bucket labels if it is an histogram with numeric labels
+func (o opMultiplyBuckets) Process(field string, value interface{}, labels common.MapStr) (string, interface{}, common.MapStr) {
```
[the remark this answers is not in the document]
Comment (jsoriano) [reply metricbeat/helper/prometheus/metric.go]: Good question, I didn't think about this, I just followed what was being done for other options.

In any case I wouldn't change it now, we can consider doing it for all options in the future.

**Comment by jsoriano** [comment]
Failing test is not related, and green in travis, merging.

**Comment by ChrsMark** [comment]
hey @jsoriano, trying to update #10585 two small questions occurred to me:
1. Shouldn't we multiply the `sum` field of the histogram too, since we have histogram's values to be scaled?
2. Why not to scale all the numeric values and only floats? If we have a histogram with 3 floats as keys and 2 integers then converting only the floats will result into a wrongly scaled histogram, right?

Sorry for not finding earlier. :disappointed: 
If we agree on those I can include them on #10585

**Comment by jsoriano** [comment]
@ChrsMark oh, you are totally right about the sum, good catch, I have opened a PR to fix this https://github.com/elastic/beats/pull/11111

Regarding the other question, it should scale anything that parses as float, what should include any number, actually the test case added here includes an integer (`1`).

Thanks!

**Pull request merged**